src/main/starlark/core/repositories/compiler.bzl

```diff
@@ -39,15 +39,27 @@ def _kotlin_capabilities_impl(repository_ctx):
         attr._artifacts_template,
         executable = False,
     )
-    template = _get_capability_template(
+    template = _get_template_by_version(
         attr.compiler_version,
         [repository_ctx.path(ct) for ct in attr._capability_templates],
+        "capabilities",
     )
     repository_ctx.template(
         "capabilities.bzl",
         template,
         executable = False,
     )
+    generated_opts_template = _get_template_by_version(
+        attr.compiler_version,
+        [repository_ctx.path(ct) for ct in attr._generated_opts_templates],
+        "generated_opts",
+    )
+    if generated_opts_template:
```

Comment:
this is not backwards compatible. Loading version sof kotlin without a generated options will fail, as the opts.kotlinc.bzl will always try to load the generated capabilities.